### PR TITLE
Propagator class for spatial superposition in z direction

### LINKED ISSUE #35: make a propagator method for spatial superposition in z direction

### PR DESCRIPTION
**Pull request by savowe**
closes #35 

working example:
```
import numpy as np
import matplotlib.pyplot as plt
import aisim as ais

pos_params = {
    'mean_x': 0.0,
    'std_x': 3.0e-3,  # cloud radius in m
    'mean_y': 0.0,
    'std_y': 3.0e-3,  # cloud radius in m
    'mean_z': 0.0,
    'std_z': 0,        # ignore z dimension, its not relevant here
}
vel_params = {
    'mean_vx': 0.0,
    # cloud velocity spread in m/s at tempearture of 3 uK
    'std_vx': ais.convert.vel_from_temp(3.0e-6),
    'mean_vy': 0.0,
    # cloud velocity spread in m/s at tempearture of 3 uK
    'std_vy': ais.convert.vel_from_temp(3.0e-6),
    'mean_vz': 0.0,
    'std_vz': ais.convert.vel_from_temp(0),
}

atoms = ais.create_random_ensemble_from_gaussian_distribution(
    pos_params, vel_params, int(1e4), state_kets=[0, 0, 0, 0, 1, 0])

center_Rabi_freq = 2*np.pi*12.5e3
r_beam = 29.5e3 / 2  # 1/e^2 beam radius in m
intensity_profile = ais.IntensityProfile(r_beam, center_Rabi_freq)
wave_vectors = ais.Wavevectors(k1=8.052945514e6, k2=-8.052802263e6)

BS = ais.SpatialSuperpositionTransitionPropagator(
    20e-6, intensity_profile=intensity_profile, wave_vectors=wave_vectors,
    n_pulses=3)
FP = ais.SpatialSuperpositionFreePropagator(230e-3, n_pulses=3)
M = ais.SpatialSuperpositionTransitionPropagator(
    40e-6, intensity_profile=intensity_profile, wave_vectors=wave_vectors,
    n_pulses=3)

atoms = BS.propagate(atoms)
atoms = FP.propagate(atoms)
atoms = M.propagate(atoms)
atoms = FP.propagate(atoms)
atoms = BS.propagate(atoms)
```

### CODE REVIEW

**Comment by bleykauf** [comment]
Flake8 complaints will now cause failed builds in Travis CI, see output for details. Also works for violations of docstring conventions.

File: tests/prop_test.py [hunk 1]
```diff
@@ -99,3 +73,53 @@ def test_two_level_transition_propagator():
...
+    init_state = [1]
+    for i in range(1, 2*n_pulses):
+        init_state.append(0)
+
+    atoms = create_random_thermal_atoms(100, state_kets=init_state)
+    atoms0 = atoms
+    # create Raman beam
+    r_beam = 29.5e-3 / 2  # 1/e^2 beam radius in m
+    wave_vectors = ais.Wavevectors(k1=8.052945514e6, k2=-8.052802263e6)
+    center_Rabi_freq = 2*np.pi/4/13.5e-15
```
Comment (savowe) [inline tests/prop_test.py:95]: the unitarity of the time evolution could not be garantued unless the rabi frequency was chosen to be very high. I suspect that has to do with the fact that in the propagate method in prop.py, first the free propagation is done and then that of the internal states. However, if you reverse the process (negative time) they will have slightly different positions.

Comment (bleykauf) [reply tests/prop_test.py:95]: Related to #54.

**Comment by bleykauf** [comment]
The tests still fail when setting the rabi frequency too low, i.e. realistic physical values, even after making the propagator symmetric. I made the tests for the propagator a little bit more convient to work with (`pi_half_time` can be used to change the Rabi frequency and the time step for the propagator).

@savowe: any ideas?

**Comment by savowe** [comment]
> 
> 
> The tests still fail when setting the rabi frequency too low, i.e. realistic physical values, even after making the propagator symmetric. I made the tests for the propagator a little bit more convient to work with (`pi_half_time` can be used to change the Rabi frequency and the time step for the propagator).
> 
> @savowe: any ideas?

fixed with dd8b25f

File: README.md [hunk 2]
```diff
@@ -43,6 +43,7 @@ notebooks](https://jupyter.org/):
     interferometry](https://github.com/bleykauf/aisim/blob/master/docs/examples/wavefront-aberrations.ipynb)
 -   [Rabi oscillations with a Gaussian beam and thermal
     atoms](https://github.com/bleykauf/aisim/blob/master/docs/examples/rabi-oscillations.ipynb)
+-   [Multiport atom interferometer](https://github.com/bleykauf/aisim/blob/master/docs/examples/multiport-ais.ipynb)
```
Comment (savowe) [inline README.md:46]: @bleykauf 
The link is wrong and should be 
https://github.com/bleykauf/aisim/blob/master/docs/examples/multiport-ai.ipynb
?

Comment (bleykauf) [reply README.md]: True, fixed.

**Pull request merged**